frontend/components/dropdown.tsx

```diff
@@ -27,8 +27,20 @@ interface DropdownMenuProps {
  * @returns {JSX.Element} A JSX element representing the Dropdown component.
  */
 
-export const Dropdown = ({ children, selected, button, buttonAlign, width, height, zIndex }: DropdownMenuProps) => {
+export const Dropdown = ({ children, selected, button, buttonAlign, width, maxHeight, zIndex }: DropdownMenuProps) => {
   const [isOpen, setIsOpen] = useState<boolean>(false);
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      const isOutsideDropdown = !(event.target as HTMLElement).closest(".dropdown");
+      if (isOutsideDropdown && isOpen) {
+        setIsOpen(false);
+      }
+    };
+
+    document.addEventListener("mousedown", handleClickOutside);
```

Comment:
Olisko näin ettei tehtäisi kovin montaa EventListener:iä
`    if (isOpen) {
      document.addEventListener("mousedown", handleClickOutside);
    }
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
`